sefaria/model/text.py

```diff
@@ -1375,14 +1366,28 @@ def _check_node_offsets(self, content, node):
 
     def _normalize(self):
         # add actualLanguage -- TODO: migration to get rid of bracket notation completely
-        actualLanguage = getattr(self, "actualLanguage", None) 
-        versionTitle = getattr(self, "versionTitle", None) 
+        actualLanguage = getattr(self, "actualLanguage", None)
+        versionTitle = getattr(self, "versionTitle", None)
         if not actualLanguage and versionTitle:
             languageCode = re.search(r"\[([a-z]{2})\]$", versionTitle)
             if languageCode and languageCode.group(1):
-                self.actualLanguage = languageCode.group(1)
+                actualLanguage = languageCode.group(1)
+        if not actualLanguage:
+            actualLanguage = self.language
+        self.actualLanguage = actualLanguage
+
+        if not getattr(self, 'languageFamilyName', None):
+            try:
+                self.languageFamilyName = constants.LANGUAGE_CODES[self.actualLanguage]
+            except KeyError:
+                self.languageFamilyName = constants.LANGUAGE_CODES[self.language]
+
+        self.isSource = getattr(self, "isSource", False)
+        if not getattr(self, "isPrimary", False):
+            if self.isSource or not VersionSet({'title': self.title}):
+                self.isPrimary = True
             else:
-                self.actualLanguage = self.language
+                self.isPrimary = False
```

Comment:
would something like this work here?
`self.isPrimary = not getattr(self, "isPrimary", False) and (self.isSource or not VersionSet({'title': self.title}))`